backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/BPartnerPrintFormatRepository.java

```diff
@@ -44,34 +52,38 @@
 @Repository
 public class BPartnerPrintFormatRepository
 {
+	@Nullable
 	public BPPrintFormat getByQuery(@NonNull final BPPrintFormatQuery bpPrintFormatQuery)
 	{
-		final I_C_BP_PrintFormat bpPrintFormatRecord = Services.get(IQueryBL.class)
+		final IQueryBuilder<I_C_BP_PrintFormat> query = Services.get(IQueryBL.class)
 				.createQueryBuilder(I_C_BP_PrintFormat.class)
 				.addEqualsFilter(I_C_BP_PrintFormat.COLUMNNAME_C_BPartner_ID, bpPrintFormatQuery.getBpartnerId().getRepoId())
-				.addEqualsFilter(I_C_BP_PrintFormat.COLUMNNAME_AD_Table_ID, bpPrintFormatQuery.getAdTableId())
-				.addEqualsFilter(I_C_BP_PrintFormat.COLUMNNAME_C_DocType_ID, bpPrintFormatQuery.getDocTypeId())
-				.addEqualsFilter(I_C_BP_PrintFormat.COLUMNNAME_AD_PrintFormat_ID, bpPrintFormatQuery.getPrintFormatId())
-				.addOnlyActiveRecordsFilter()
-				.create()
-				.firstOnlyOrNull(I_C_BP_PrintFormat.class);
+				.addInArrayFilter(I_C_BP_PrintFormat.COLUMNNAME_AD_Table_ID, bpPrintFormatQuery.getAdTableId(), null)
+				.addInArrayFilter(I_C_BP_PrintFormat.COLUMNNAME_C_DocType_ID, bpPrintFormatQuery.getDocTypeId(), null)
+				.addInArrayFilter(I_C_BP_PrintFormat.COLUMNNAME_C_BPartner_Location_ID, bpPrintFormatQuery.getBPartnerLocationId(), null)
+				.addInArrayFilter(I_C_BP_PrintFormat.COLUMNNAME_AD_PrintFormat_ID, bpPrintFormatQuery.getPrintFormatId(), null)
+				.addOnlyActiveRecordsFilter();
 
-		if (bpPrintFormatRecord == null)
+		if(bpPrintFormatQuery.isOnlyCopiesGreaterZero())
 		{
-			return null;
+			query.addCompareFilter(I_C_BP_PrintFormat.COLUMNNAME_DocumentCopies_Override, CompareQueryFilter.Operator.GREATER, 0);
```

Comment:
wondering if we shall also add query.addIsNotNull ....

Reply:
not needed, column is mandatory